pandas/tseries/tests/test_timeseries.py

```diff
@@ -2454,6 +2454,61 @@ def test_intersection_bug_1708(self):
+    def test_datetime64_with_DateOffset(self):
+        for klass in (Series, DatetimeIndex):
+            s = klass(date_range('2000-01-01', '2000-01-31'))
+            result = s + pd.DateOffset(years=1)
+            result2 = pd.DateOffset(years=1) + s
+            exp = klass(date_range('2001-01-01', '2001-01-31'))
+            self.assert_numpy_array_equal(result, exp)
+            self.assert_numpy_array_equal(result2, exp)
+
+            result = s - pd.DateOffset(years=1)
+            exp = klass(date_range('1999-01-01', '1999-01-31'))
+            self.assert_numpy_array_equal(result, exp)
+
+            s = klass([Timestamp('2000-01-15 00:15:00', tz='US/Central'),
+                       pd.Timestamp('2000-02-15', tz='US/Central')])
+            result = s + pd.offsets.MonthEnd()
+            result2 = pd.offsets.MonthEnd() + s
+            exp = klass([Timestamp('2000-01-31 00:15:00', tz='US/Central'),
+                         Timestamp('2000-02-29', tz='US/Central')])
+            self.assert_numpy_array_equal(result, exp)
+            self.assert_numpy_array_equal(result2, exp)
+
+            #DateOffset relativedelta fastpath
+            relative_kwargs = [('years', 2), ('months', 5), ('days', 3),
+                            ('hours', 5), ('minutes', 10), ('seconds', 2),
+                            ('microseconds', 5)]
+            for i, kwd in enumerate(relative_kwargs):
+                off = pd.DateOffset(**dict([kwd]))
+                self.assert_numpy_array_equal(s.map(lambda x: x + off), s + off)
+                self.assert_numpy_array_equal(s.map(lambda x: x - off), s - off)
+                off = pd.DateOffset(**dict(relative_kwargs[:i+1]))
+                self.assert_numpy_array_equal(s.map(lambda x: x + off), s + off)
+                self.assert_numpy_array_equal(s.map(lambda x: x - off), s - off)
+            #valid non-tick Offsets
+            for do in ['BusinessDay', 'BDay', 'CustomBusinessDay', 'CDay',
+                    'CBMonthEnd','CBMonthBegin', 'MonthBegin', 'BMonthBegin',
+                    'MonthEnd', 'BMonthEnd', 'BusinessHour', 'YearBegin',
+                    'BYearBegin', 'YearEnd', 'BYearEnd', 'QuarterBegin',
+                    'BQuarterBegin', 'QuarterEnd', 'BQuarterEnd', 'Week',
+                    ('LastWeekOfMonth', {'weekday':2}),
+                    ('FY5253Quarter', {'qtr_with_extra_week': 1, 'startingMonth': 1,
+                                        'weekday': 2, 'variation': 'nearest'}),
+                    ('FY5253',{'weekday': 0, 'startingMonth': 2, 'variation': 'nearest'}),
+                    ('WeekOfMonth', {'weekday': 2, 'week': 2}),
+                    'Easter']:
+                if isinstance(do, tuple):
+                    do, kwargs = do
+                else:
+                    do = do
+                    kwargs = {}
+                op = getattr(pd.offsets,do)(5, **kwargs)
+                self.assert_numpy_array_equal(s.map(lambda x: x + op), s + op)
```

Comment:
these are going to show the ``PerformanceWarning`` yes?, wrap an ``assert_produces_warning(....)`` around whichever shows